generator/main.go

```diff
@@ -46,21 +47,22 @@ func generateConfig(nodes *Node, nameToNode map[string]*Node) {
 		log.Fatalf("Error parsing generated config: %s", err)
 	}
 
-	f, err := os.Create("snmp.yml")
+	f, err := os.Create(outputPath)
 	if err != nil {
 		log.Fatalf("Error opening output file: %s", err)
 	}
 	_, err = f.Write(out)
 	if err != nil {
 		log.Fatalf("Error writing to output file: %s", err)
 	}
-	log.Infof("Config written to snmp.yml")
+	log.Infof("Config written to %s", outputPath)
 }
 
 var (
 	generateCommand    = kingpin.Command("generate", "Generate snmp.yml from generator.yml")
 	parseErrorsCommand = kingpin.Command("parse_errors", "Debug: Print the parse errors output by NetSNMP")
 	dumpCommand        = kingpin.Command("dump", "Debug: Dump the parsed and prepared MIBs")
+	outputDir          = kingpin.Flag("output-dir", "Directory to write resulting config file").Default("/opt").Short('o').String()
```

Comment:
This is a breaking change, it should continue to write to the current directory.

It's also usual to take in the full path.